File: Easy/Sum of Digits of String After Convert/getLucky.java
class Solution {
    public int getLucky(String s, int k) {
        StringBuilder numStr = new StringBuilder();

        for (char c : s.toCharArray()) {
            int position = c - 'a' +1;
            numStr.append(position);
        }
        int result = 0;

        for (char digit: numStr.toString().toCharArray()) {
            result += Character.getNumericValue(digit);
        }
        for (int i = 1; i < k; i++) {
            int newResult =0;
            while (result >0) {
                newResult += result %10;
                result /= 10;
            }
            result = newResult;
        }
        return result;
    }
}

//QED
//Problem 1945 (Daily Problem: Sum of Digits of String After Convert) - Jason Balayev (python)   

File: Easy/Sum of Digits of String After Convert/getLucky.py
class Solution(object):
    def getLucky(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        num_str = ''.join(str(ord(char) - ord('a') +1) for char in s)
        result = sum(int(digit) for digit in num_str)
        for i in range(1, k):
            result = sum(int(digit) for digit in str(result))
        return result
    
#QED
#Problem 1945 (Daily Problem: Sum of Digits of String After Convert) - Jason Balayev (python)   